### DISCUSSION TITLE
Problems using .extra property

### DISCUSSION THREAD

**Discussion by FelixBetz** [ics-py/ics-py#371] [Q&A] [answered]
Hi,
first of all I want to say thank you for the amazing work you did with ics-py

I have a problem adding a custom property to an event with `.extra`

 **I stuck to the documentation:** 
[https://icspy.readthedocs.io/en/stable/advanced.html#low-level-api](https://icspy.readthedocs.io/en/stable/advanced.html#low-level-api)
```
e.name == "Name of the event"
e.extra == [ContentLine(name="FOO", value="BAR")]
```
**The  `==`  is an documentation mistake, isn't it?** 

**What i did:** 
```
#create new calendar
ics_calendar = Calendar()

#create new event
event = Event()
event.extra = [ContentLine(name="FOO", value="BAR")]

#add event to calendar
ics_calendar .events.add(event)

#write calendar to ics file
with open("calendar.ics", 'w',  encoding="utf-8") as ics_file:
   ics_file.writelines(ics_calendar .serialize_iter())
```

**when I run these lines of code I get the following error:**

```
  File "main.py", line 163, in <module>generate_calenders()
  File "main.py", line 155, in generate_calenders  ics_file.writelines(ics_calender.serialize_iter())
  File "......\ics\component.py", line 81, in serialize_iter  return self.serialize().splitlines(keepends=True)
  File "......\ics\component.py", line 72, in serialize   output(self, container)
  File "......\ics\serializers\icalendar_serializer.py", line 27, in serialize_event  container.append(event.serialize())
  File "......\ics\component.py", line 70, in serialize   container = self.extra.clone()
  AttributeError: 'list' object has no attribute 'clone'
```
Am I doing something wrong? Can someone help me?
Many thanks in advance
Felix

**Comment by N-Coder** [comment, marked as answer]
Overwriting `.extra`, which is a `list` subclass called `ics.grammar.parse.Container` and that, amongst others, provides the `clone()` method, unfortunately removes this method. So the `.extra`s should only ever be appended/removed/cleared, but never entirely replaced. The correct code should be:
```python
event.extra.append(ContentLine(name="FOO", value="BAR"))
```

We should probably document this and also make sure to prevent this bug (or at least get sensible error messages in this case) as this probably also happens with the 0.8-alpha. And this also kind of shows that the coverage of the v0.7 test-suite has potential for improval, which is also good to keep in mind when working on porting it to 0.8.

**Comment by FelixBetz** [comment]
Thank you very much @N-Coder !
That solved my problem!
